HHS/simpler-grants-gov
# [Issue #3078] cleanup translation related code

## Summary
Fixes #3078
### Time to review: __30 mins__

## Changes proposed

## Context for reviewers
When testing translation behavior, it was difficult to track down why the Spanish version of the site was not loading. In the end, it was a matter of us manually setting "en" as the locale in a number of places unnecessarily. Since `getMessagesWithFallbacks` handles filling any missing messages in with English, so there isn't a need to manually set 
the request locale to "en".

This change updates the entire app to support dynamically set locales.

It also:

* updates outdated next intl `set locale` methods (unstable_setRequestLocale is now stable, see https://github.com/amannn/next-intl/pull/1437)
* includes a couple of quality of life improvements related to translation code

### Test steps

1. add some spanish messages for local testing. Add the following to the spanish messages file:
```
export const messages = {
  ErrorPages: {
    page_not_found: {
      title: "No pagina nunca",
    },
  },
  Subscribe: {
    form: {
      name: "Nombre Primero",
      lastName: "Nombre Ultimo",
    },
  },
};
```
2. visit http://localhost:3000/es/subscribe
3. _VERIFY_: the spanish input titles added in step one appear
2. visit http://localhost:3000/es/subscrdfafdfibe
3. _VERIFY_: the spanish heading added in step one appear


## Additional information
> Screenshots, GIF demos, code examples or output to help show the changes working as expected.
